### PR TITLE
Support Python Launcher on Windows when using `poetry env use` and `poetry env remove` (#3520)

### PR DESCRIPTION
**Pull request by sitd0813**
# Pull Request Check List

Resolves: #issue-number-here

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

Fix <https://github.com/python-poetry/poetry/issues/3520>

~~I did not add any test because I'm not really familiar with PyTest and thus had no idea how to add Windows specific test. However, the code was built and tested on Windows 10 device and it proved to resolve <https://github.com/python-poetry/poetry/issues/3520> without apparent errors.~~

EDIT

Added test to verify if `EnvManager._execute_python_script` works as expected.

### CODE REVIEW

**Comment by sitd0813** [comment]
Both tests currently failed due to `ConnectionError`. I have no idea why it occurred. It seems like there is an issue with the testing platform.

File: poetry/utils/env.py [hunk 1]
```diff
@@ -448,22 +461,14 @@ def activate(self, python: str, io: IO) -> "Env":
...
-                            "-c",
-                            "\"import sys; print('.'.join([str(s) for s in sys.version_info[:3]]))\"",
-                        ]
-                    ),
-                    shell=True,
-                )
-            )
+            python_version = execute_python_script(
+                python,
+                "\"import sys; print('.'.join([str(s) for s in sys.version_info[:3]]))\"",
```
Comment (Leonardofreua) [inline poetry/utils/env.py:466]: What do you think about creating a constant for the script `"\"import sys; print('.'.join([str(s) for s in sys.version_info[:3]]))\""`?

Comment (sitd0813) [reply poetry/utils/env.py]: Thank you for reviewing my code. I added the constant for the script. Also, I made the `execute_python_script` a member function of `EnvManager` as it is only used inside it. It is now `EnvManager._execute_python_script`.

Comment (Leonardofreua) [reply poetry/utils/env.py]: Great. Good catch.

Comment (sitd0813) [reply poetry/utils/env.py]: Is there anything else I need to do? It seems that learning more test require approval.

Comment (Leonardofreua) [reply poetry/utils/env.py]: I think it needs more tests and one more reviewer.

**Comment by dalito** [comment]
As a fan of the launcher I have tested your PR with poetry-1.2.0a2 on Win10. It mostly works. 👍  

Only when removing an env an error occurs. The env is removed nevertheless. The error is probably due to a "feature" in Win10. When you don't have python on path but enter "python", Win10 suggests to install Python from the Microsoft store. So testing for "python" gives an unexpected output. Maybe it is better to test for the presence of "pip"?

```
λ poetry env use python3.8
Creating virtualenv mared-NfQ5kqas-py3.8 in C:\Users\zzz\AppData\Local\pypoetry\Cache\virtualenvs
Using virtualenv: C:\Users\zzz\AppData\Local\pypoetry\Cache\virtualenvs\mared-NfQ5kqas-py3.8

λ poetry env remove python3.8

  CalledProcessError

  Command 'python -W ignore -' returned non-zero exit status 9009.

  at C:\dev\Python39\lib\subprocess.py:528 in run
       524│             # We don't call process.wait() as .__exit__ does that for us.
       525│             raise
       526│         retcode = process.poll()
       527│         if check and retcode:
    →  528│             raise CalledProcessError(retcode, process.args,
       529│                                      output=stdout, stderr=stderr)
       530│     return CompletedProcess(process.args, retcode, stdout, stderr)
       531│
       532│

The following error occurred when trying to handle this error:


  EnvCommandError

  Command python -W ignore - errored with the following return code 9009, and output:
  Python was not found but can be installed from the Microsoft Store: https://go.microsoft.com/fwlink?linkID=2082640input was : import sys

  if hasattr(sys, "real_prefix"):
      print(sys.real_prefix)
  elif hasattr(sys, "base_prefix"):
      print(sys.base_prefix)
  else:
      print(sys.prefix)


  at ~\_poetry\venv\lib\site-packages\poetry\utils\env.py:1285 in _run
      1281│                 output = subprocess.check_output(
      1282│                     cmd, stderr=subprocess.STDOUT, env=env, **kwargs
      1283│                 )
      1284│         except CalledProcessError as e:
    → 1285│             raise EnvCommandError(e, input=input_)
      1286│
      1287│         return decode(output)
      1288│
      1289│     def execute(self, bin: str, *args: str, **kwargs: Any) -> Optional[int]:
```

**Comment by neersighted** [comment]
The current consensus is that we should use `pythonfinder` instead of trying to write out this logic ourselves. Feel free to re-open this PR if you'd like to switch to that approach -- otherwise, @bmarroquin has expressed interest in PRing `pythonfinder` support.

**Pull request closed without merging**